modules/auxiliary/scanner/misc/dahua_dvr_auth_bypass.rb

```diff
@@ -166,7 +167,7 @@ def grab_nas
     if /[\x00]{16,}(?<ftpuser>[[:print:]]+)[\x00]{16,}(?<ftppass>[[:print:]]+)/ =~ data
       ftpuser.strip!
       ftppass.strip!
-      unless ftpuser.blank? || ftppass.blank?
+      if !ftpuser.blank? || !ftppass.blank?
```

Comment:
Yeah, I think both the old logic and the suggested logic are wrong.  Unless the protocol in question is just terrible, there should never be a case where that regex matches and `ftpuser` or `ftppass` is blank.  In both the old and suggested logic, it seems like there is this edge case where one of the values might be blank.  I'd suggest going one of two ways:

1. Remove this `if`/`unless` logic that is trying to handle blank `ftpuser` or `ftppass`
2. Modify the two relevant `print_good` lines below to only print `ftpuser` or `ftppass` if non-blank, as appropriate.